src/Analysis/Ast/Impl/Modules/PythonModule.cs

```diff
@@ -313,7 +313,16 @@ internal PythonModule(ModuleCreationOptions creationOptions, IServiceContainer s
             _linkedParseCts = CancellationTokenSource.CreateLinkedTokenSource(_disposeToken.CancellationToken, _parseCts.Token);
 
             ContentState = State.Parsing;
-            _parsingTask = Task.Run(() => Parse(_linkedParseCts.Token), _linkedParseCts.Token);
+            _parsingTask = Task.Run(() => ParseAndLogExceptions(_linkedParseCts.Token), _linkedParseCts.Token);
+        }
+
+        private void ParseAndLogExceptions(CancellationToken cancellationToken) {
+            try {
+                Parse(cancellationToken);
+            } catch (Exception ex) when (!(ex is OperationCanceledException)) {
```

Comment:
Other option is to `catch (OperationCanceledException) { throw; } catch (Exception ex) { ... }`, but I'm not a .NET style expert.